scm-webapp/src/test/java/sonia/scm/api/v2/resources/BrowserResultToFileObjectDtoMapperTest.java

```diff
@@ -115,18 +114,36 @@ public void shouldSetLinksCorrectly() {
     assertThat(dto.getLinks().getLinkBy("self").get().getHref()).contains("path");
   }
 
+  @Test
+  public void shouldEncodeFileLinks() {
+    BrowserResult browserResult = new BrowserResult("Revision", createFileObject("c:file"));
+    NamespaceAndName namespaceAndName = new NamespaceAndName("foo", "bar");
+
+    FileObjectDto dto = mapper.map(browserResult, namespaceAndName, 0);
+
+    assertThat(dto.getLinks().getLinkBy("self").get().getHref()).isEqualTo("http://example.com/base/v2/repositories/foo/bar/content/Revision/c:file");
+  }
+
   private BrowserResult createBrowserResult() {
-    return new BrowserResult("Revision", createFileObject());
+    return new BrowserResult("Revision", createDirectoryObject());
+  }
+
+  private FileObject createDirectoryObject() {
+    FileObject directory = new FileObject();
+    directory.setName("");
+    directory.setPath("/path");
+    directory.setDirectory(true);
+
+    directory.addChild(fileObject1);
+    directory.addChild(fileObject2);
+    return directory;
   }
 
-  private FileObject createFileObject() {
+  private FileObject createFileObject(String name) {
```

Comment:
```suggestion
  private FileObject createFileObject(String path) {
```

Reply:
But don't we say "The file has the name xyz" and not "The file has the path xyz"?